tests/test_datastructures.py

```diff
@@ -88,6 +88,7 @@ def test_basic_interface(self):
 
         # simple getitem gives the first value
         assert md["a"] == 1
+        assert {**md}["a"] == 1
```

Comment:
Worth adding a comment here about the unexpected `**` regression that this is aiming to keep from happening again? Someone in the future might delete this line thinking it’s not actually adding additional coverage over the previous line.

And if we can find a way to write a regression test in terms of just the specific `__iter__` method we’re changing here (in addition to the composite operation that `**` exercises), that would really show what’s going on better.